tests/server.rs

```diff
@@ -498,7 +498,7 @@ fn head_response_doesnt_send_body() {
     let mut response = String::new();
     req.read_to_string(&mut response).unwrap();
 
-    assert!(response.contains("content-length: 11\r\n"));
+    assert!(!has_header(&response, "content-length"));
```

Comment:
I believe we want to keep this behavior. It's correct and expected that a `HEAD` request will be able to check what the `content-length` *would be* if sending a `GET`.

Reply:
You are very correct. I have now reverted this test and made sure the right behavior is followed for HEAD requests.